api/collections/serializers.py

```diff
@@ -94,3 +96,72 @@ def get_absolute_url(self, obj):
                 'node_link_id': obj._id
             }
         )
+
+
+class LinkedNode(JSONAPIRelationshipSerializer):
+    class Meta:
+        type_ = 'linked_nodes'
+
+
+class CollectionLinkedNodesRelationshipSerializer(ser.Serializer):
+
+    data = ser.ListField(child=LinkedNode())
+    links = LinksField({'self': 'get_self_url',
+                        'html': 'get_related_url'})
+
+    def get_self_url(self, obj):
+        return obj['self'].linked_nodes_self_url
+
+    def get_related_url(self, obj):
+        return obj['self'].linked_nodes_related_url
+
+    class Meta:
+        type_ = 'linked_nodes'
+
+    def get_pointers_to_add_remove(self, pointers, new_pointers):
+        diff = relationship_diff(
+            current_items={pointer.node._id: pointer for pointer in pointers},
+            new_items={val['node']['_id']: val for val in new_pointers}
+        )
+
+        nodes_to_add = []
+        for node_id in diff['add']:
+            node = Node.load(node_id)
+            if not node:
+                raise exceptions.NotFound
```

Comment:
This should have a descriptive error message so that the client knows which node can't be found.